src/components/va-text-input/va-text-input.tsx

```diff
@@ -120,6 +130,15 @@ export class VaTextInput {
     const describedBy =
       `${this.ariaDescribedby} ${this.error ? 'error-message' : ''}`.trim() ||
       null; // Null so we don't add the attribute if we have an empty string
+    const inputMode =
+      this.inputmode ?
+      this.inputmode :
+      null; // Null so we don't add the attribute if we have an empty string
```

Comment:
Should this also be filtered by an allowed list?

Reply:
I went back and forth about this, but decided not to filter because we're not trying to restrict the allowed values in the same way as with `type`, (Unless perhaps we wanted to disallow passing `none`?) and browsers default to using `text` if an invalid value is passed.  That said I don't have strong feelings about it if you'd like to see it there.